datafusion/datasource/src/file_groups.rs

```diff
@@ -354,6 +361,115 @@ impl FileGroupPartitioner {
     }
 }
 
+/// Represents a group of partitioned files that'll be processed by a single thread.
+/// Maintains optional statistics across all files in the group.
+#[derive(Debug, Clone)]
+pub struct FileGroup {
+    /// The files in this group
+    pub files: Vec<PartitionedFile>,
+    /// Optional statistics for all files in the group
+    pub statistics: Option<Statistics>,
```

Comment:
We can compute the `statistics` in the method of `ListingTable` 

https://github.com/apache/datafusion/blob/5210a2bac32e43dc7bf6e7e6000cdeaf2833c06e/datafusion/core/src/datasource/listing/table.rs#L1086-L1091

Reply:
I recommend:
1. Remove the `pub` from the fields (so we can potentially change the representation later)
2. Add a `into_inner()` method that returns the inner Vec

Reply:
done: https://github.com/apache/datafusion/pull/15379/commits/7415f7c6846b33136ee0c833c6836a3c97508b6b